### ISSUE TITLE
Enrich flows based on non-primary Pod IPs

### ISSUE THREAD

**Issue by vakalapa** [microsoft/retina#224]
**Is your feature request related to a problem? Please describe.**
Currently retina only enriches flows based on the primary IP of the Pod, since we only add primary IP to the `ipToEpKey` map in cache, we should support adding all IPs so we can get the EP for secondary IPs as well.

https://github.com/microsoft/retina/blob/5cfb7efcc24b7fa48ff951e9f560c3ebe58f5af3/pkg/controllers/cache/cache.go#L204-L208

**Comment by FZhg** [comment]
Hi @vakalapa ,
I would like to work on this issue. Here are my initial thoughts:
1. I shall use the `GetIPs` method of  `IPAddresses`  struct to get all the IPs for the endpoint. 
2.  Change the following line to add the ips into the caches, with the same endpoint key.
https://github.com/microsoft/retina/blob/5cfb7efcc24b7fa48ff951e9f560c3ebe58f5af3/pkg/controllers/cache/cache.go#L222C1-L222C28


I believe once I have completed the above 2 steps, the `enricher` will automatically enriches the flows with secondary IPs as well. Could you confirm this? Thanks.

**Comment by vakalapa** [comment]
@FZhg  Looks like you closed your PR, are you still thinking to work on this issue ?

**Comment by FZhg** [comment]
> @FZhg Looks like you closed your PR, are you still thinking to work on this issue ?

Yes, I am! I realized that If I create a PR draft, it will be a duplicate for this issue on the triage board. That's why I closed my PR.
I am planning to ask questions tomorrow at the office hours.

I am thinking about whether I should use a concurrent hash map to solve the secanrio you mentioned. I am hesitating because of the overhead it would introduce.
